src/web/blocks.rs

```diff
@@ -52,41 +54,49 @@ pub struct BlockDTO {
+    parent_block_hash: Option<HashOf<VersionedCommittedBlock>>,
+    transactions_merkle_root_hash: Option<HashOf<MerkleTree<VersionedSignedTransaction>>>,
+    rejected_transactions_merkle_root_hash: Option<HashOf<MerkleTree<VersionedSignedTransaction>>>,
+    // invalidated_blocks_hashes: Vec<SerScaleHex<Hash>>,
+    transactions: Vec<TransactionValue>,
+    // rejected_transactions: Vec<SerScaleHex<VersionedSignedTransaction>>,
+    // view_change_proofs: Vec<SerScaleHex<Hash>>,
 }
 
-impl TryFrom<BlockValue> for BlockDTO {
+impl TryFrom<VersionedCommittedBlock> for BlockDTO {
     type Error = color_eyre::Report;
 
-    fn try_from(block: BlockValue) -> Result<Self> {
+    fn try_from(block: VersionedCommittedBlock) -> Result<Self> {
+        let committed_block = block.into_v1();
+        // the querybox output is  VersionedCommittedBlock -> committedBlock
         Ok(Self {
-            height: block.header.height.try_into()?,
-            timestamp: Timestamp::try_from(block.header.timestamp)?,
-            block_hash: block.header.current_block_hash.into(),
-            parent_block_hash: block.header.previous_block_hash.into(),
-            transactions_merkle_root_hash: block.header.transactions_hash.into(),
-            rejected_transactions_merkle_root_hash: block.header.rejected_transactions_hash.into(),
-            invalidated_blocks_hashes: block
+            height: committed_block.header.height.try_into()?,
+            timestamp: Timestamp::try_from(committed_block.header.timestamp)?,
+            block_hash: committed_block.hash().into(),
+            parent_block_hash: committed_block.header.previous_block_hash,
+            transactions_merkle_root_hash: committed_block.header.transactions_hash,
+            rejected_transactions_merkle_root_hash: committed_block
                 .header
-                .invalidated_blocks_hashes
-                .into_iter()
-                .map(Into::into)
-                .collect(),
-            transactions: block.transactions.into_iter().map(Into::into).collect(),
-            rejected_transactions: block
-                .rejected_transactions
+                .rejected_transactions_hash,
+            // invalidated_blocks_hashes:  committed_block
+            //     .header
+            //     .invalidated_blocks_hashescl
+            //     .into_iter()
+            //     .map(Into::into)
+            //     .collect(),
+            transactions: committed_block
+                .transactions
                 .into_iter()
                 .map(Into::into)
                 .collect(),
+            // rejected_transactions:  committed_block
+            //     .rejected_transactions
+            //     .into_iter()
+            //     .map(Into::into)
+            //     .collect(),
 
             // FIXME https://github.com/hyperledger/iroha/issues/2277
-            view_change_proofs: Vec::new(),
+            // view_change_proofs: Vec::new(),
```

Comment:
Could you enumerate pros and cons of removal of `view_change_proofs` (which is empty for now) from the public API?